### DISCUSSION TITLE
How to show the content of a shared md-file in a layout file?

### DISCUSSION THREAD

**Discussion by markushuebner** [datenstrom/community#877] [Ask a question] [answered]
Hi all,
following situation:

_yellow
--content
----2-test
------page.md
----shared
------info.md_

to show the content of the _page.md_ one uses within _default.html_

`<?php echo $this->yellow->page->getContentHtml() ?>`

I found [https://github.com/datenstrom/yellow/discussions/848](https://github.com/datenstrom/yellow/discussions/848) so

that one
`<?php echo $this->yellow->content->find("/shared/")->getContent() ?>`
only works with a file _page.md_ located within the folder _shared_. But, what if I would like to use that specific _info.md_?
`<?php echo $this->yellow->content->find("/shared/")->getPage("info")->getContent(); ?>`
or
`<?php echo $this->yellow->content->find("/shared/")->getPage("info")->getContentHtml(); ?>`
do not end with an error message but deliver just "blanc" - what of course is not what I am looking for.

I appreciate any suggestions. Thank you!

Greetings,

Markus

**Comment by markseuffert** [comment, marked as answer]
Hello Markus. The class [YellowPage](https://datenstrom.se/yellow/help/api-for-developers#yellow-page) let's you access shared pages.

Here's an example layout file `system/layouts/default.html`:

```
<?php $this->yellow->layout("header") ?>
<div class="content">
<div class="main" role="main">
<h1><?php echo $this->yellow->page->getHtml("titleContent") ?></h1>
<?php echo $this->yellow->page->getPage("info")->getContentHtml() ?>
<?php echo $this->yellow->page->getContentHtml() ?>
</div>
</div>
<?php $this->yellow->layout("footer") ?>
```

Here's an example content file `content/shared/info.md`:

```
---
Title: Info
Status: shared
---
!!! This website is currently under reconstruction. 
!!! If something looks wrong [contact our webmaster](/contact/). 
```

There are possibly other ways to do this with the API. Please note that shared content files need `Status: shared` at the top of a page. The core checks this in `parseMetaDataShared()` when making a list of shared pages. Hope this helps and happy coding.

**Comment by markushuebner** [comment]
Thanks a lot Mark!

> Please note that shared content files need Status: shared at the top of a page. 

THAT was the point I missed! I just started with yellow and bäm, fell into a newbie-trap...

**Reply by markseuffert** [reply]
Thanks a lot for your feedback. By the way, I have thought multiple times about removing that page setting. It dates back to a time when you could have shared pages in any content folder, not just in the shared content folder. Maybe it's time to remove that page setting, at the moment it's there for performance reasons. Something to discuss in the next API review.

**Reply by markseuffert** [reply]
Done, that setting is no longer needed in latest core. Shared content files automatically have `Status: shared`.

**Comment by markushuebner** [comment]
Thanks a lot! 🙂Gesendet von Outlook für Android
